graphql_compiler/fast_introspection.py

```diff
@@ -0,0 +1,162 @@
+"""Module to execute GraphQL introspection query faster than graphql-core.
+
+On a large schema with tens of thousands of types, the approach using graphql-core's
+graphql_sync takes 26 seconds to run, whereas this module executes the same query in
+1.37 seconds. This module calls the same Type resolvers as graphql_sync to prevent
+code duplication with graphql-core.
+
+This module is fully dependent on the introspection query returned by `get_introspection_query`
+from graphql-core (with default parameters). Changes to that query will need to be
+reflected in this module. This module also assumes that the introspection query can
+be executed on the given schema.
+"""
+
+
+from typing import Any, Dict, Tuple, cast
+
+from graphql import (
+    GraphQLAbstractType,
+    GraphQLArgument,
+    GraphQLDirective,
+    GraphQLEnumType,
+    GraphQLEnumValue,
+    GraphQLField,
+    GraphQLNamedType,
+    GraphQLObjectType,
+    GraphQLSchema,
+    GraphQLType,
+    introspection_types,
+    is_abstract_type,
+)
+
+
+__Schema = cast(GraphQLObjectType, introspection_types["__Schema"])
+__Directive = cast(GraphQLObjectType, introspection_types["__Directive"])
+__DirectiveLocation = cast(GraphQLEnumType, introspection_types["__DirectiveLocation"])
+__Type = cast(GraphQLObjectType, introspection_types["__Type"])
+__Field = cast(GraphQLObjectType, introspection_types["__Field"])
+__InputValue = cast(GraphQLObjectType, introspection_types["__InputValue"])
+__EnumValue = cast(GraphQLObjectType, introspection_types["__EnumValue"])
+__TypeKind = cast(GraphQLEnumType, introspection_types["__TypeKind"])
```

Comment:
`pylint` is wrong here, `introspection_types` is subscriptable since it's of type `FrozenDict`, a subtype of `Dict`.
```suggestion
# pylint: disable=unsubscriptable-object
__Schema = cast(GraphQLObjectType, introspection_types["__Schema"])
__Directive = cast(GraphQLObjectType, introspection_types["__Directive"])
__DirectiveLocation = cast(GraphQLEnumType, introspection_types["__DirectiveLocation"])
__Type = cast(GraphQLObjectType, introspection_types["__Type"])
__Field = cast(GraphQLObjectType, introspection_types["__Field"])
__InputValue = cast(GraphQLObjectType, introspection_types["__InputValue"])
__EnumValue = cast(GraphQLObjectType, introspection_types["__EnumValue"])
__TypeKind = cast(GraphQLEnumType, introspection_types["__TypeKind"])
# pylint: enable=unsubscriptable-object
```
Feel free to open an issue in https://github.com/PyCQA/pylint about this, if there isn't one already — this is a pretty good reproducible example of a bug.